### PR TITLE
brings wiremods up to date

### PR DESCRIPTION
**Pull request by jupyterkat**
## About The Pull Request

brings wiremods up to date
pr list goes here

## Why It's Good For The Game

wiremod is currently outdated and has rare uses

## Changelog
:cl: mothblocks, watermelon914
add: updated wiremods
add: quantum pads has a usb port now
add: crew monitoring consoles has a usb port now
add: teleporters has a usb port now
add: air alarms has a usb port now
add: bluespace launchpads has a usb port now
add: digital valves has a usb port now
add: conveyor belts switch has a usb port now
add: several more circuit components
/:cl:

### CODE REVIEW

File: code/modules/wiremod/components/admin/proccall.dm [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * # Proc Call Component
+ *
+ * A component that calls a proc on an object and outputs the return value
+ */
+/obj/item/circuit_component/proccall
+	display_name = "Proc Call"
+	desc = "A component that calls a proc on an object."
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL|CIRCUIT_FLAG_ADMIN
```
Comment (francinum) [inline code/modules/wiremod/components/admin/proccall.dm:13]: No. What the *FUCK ARE YOU THINKING*

**Review by francinum — CHANGES_REQUESTED** [review]

File: code/modules/wiremod/components/admin/proccall.dm (hunk 1 above)
Comment (ike709) [reply code/modules/wiremod/components/admin/proccall.dm:13]: ...did TG actually merge this or did you make it?

Comment (francinum) [reply code/modules/wiremod/components/admin/proccall.dm:13]: Apparently they did?

Still incredibly goddamn dangerous even if the circuits *are* protected.
This is *COMPLETELY UNMENTIONED IN THE PR BODY*

**Comment by ike709** [comment]
Remove every admin circuit.

File: code/modules/wiremod/components/admin/proccall.dm (hunk 1 above)
Comment (Archanial) [reply code/modules/wiremod/components/admin/proccall.dm:13]: its admin only calm down

Comment (ike709) [reply code/modules/wiremod/components/admin/proccall.dm:13]: It's still a terrible idea.

**Comment by jupyterkat** [comment]
no fabs can build the goddamn admin circuits except for the debug fab dammit. 

Incredibly dangerous? Just fucking revoke all admins from proc calling then. Go fuck off franc

If tgmins can use it for events, then I don’t see why we can’t

File: code/modules/wiremod/components/admin/proccall.dm (hunk 1 above)
Comment (jupyterkat) [reply code/modules/wiremod/components/admin/proccall.dm:13]: no it’s not, admins are allowed to proc call anytime

**Comment by ike709** [comment]
Remove every admin circuit.

**Comment by jupyterkat** [comment]
no

**Comment by Sinestia** [comment]
You should remove things when the headcoder asks but I don't see the issue with this

Is it dangerous? Of course, but so are a lot of other things admins have. 

 I think beemins should be trustable enough to see a component that literally does arbitrary proccalls as being dangerous, and like any crashing the server with idiocy you'd be held responsible. Oh well.

File: code/modules/wiremod/components/admin/proccall.dm (hunk 1 above)
Comment (Archanial) [reply code/modules/wiremod/components/admin/proccall.dm:13]: It's not terrible idea, how is it different from admin calling sdql or proccall?

**Comment by mcmeiler** [comment]
It's still admin only. If it ever were to fall into the hands of a normal user, who then proceeds to crash the server, it's on the admin who spawned the circuit *and* the user who crashed the server. 
Though this requires an admin to spawn it first, which shouldn't be happening in the first place.

**Comment by Watermelon914** [comment]
don't think you guys realise but players cannot even use these in normal circuits because the integrated circuit needs to be set to admin_only for any of the admin related circuit components to work. Not only that, when an integrated circuit is set to admin_only, they become inaccessible to regular players and regular players are no longer able to interact with it at all.

Sounds like you don't trust admins to use event tooling at all.

**Comment by ike709** [comment]
@Watermelon914  I don't trust admins with a privilege elevation exploit. Any admin with VV can use these circuits to do arbitrary SDQL calls without actually checking if they have `R_DEBUG`, to give an example of why this is a bad idea.

**Comment by jupyterkat** [comment]
it checks for varedit, if you have varedit you can mass edit shit already

**Comment by ike709** [comment]
The existing permissions system being a mess is not an excuse to make it more irrelevant.

**Comment by ike709** [comment]
There's not an argument anyone can make that will get me to merge this PR in its current state, so don't waste the effort trying.

**Comment by jupyterkat** [comment]
then you can just add r_debug to the required perms

**Comment by jupyterkat** [comment]
> The existing permissions system being a mess is not an excuse to make it more irrelevant.

go fix it then

**Comment by Watermelon914** [comment]
> @Watermelon914 I don't trust admins with a privilege elevation exploit. Any admin with VV can use these circuits to do arbitrary SDQL calls without actually checking if they have `R_DEBUG`, to give an example of why this is a bad idea.

I was assuming beestation trusted their admins to not abuse the permissions given to them. If that's not the case, then fair enough.

On tgstation, this isn't an issue because most ranks have access to R_DEBUG. If you don't give admins any debug permissions, then adding admin circuits to beestation is a bad idea because they are only designed for event tooling and nothing else. In order to be effective in any event you run, it practically requires you to use proccalls.

There is no reason to add an event tool if it is going to be locked behind debug.

**Comment by Watermelon914** [comment]
However, I recommend you port the system but not the components that come with it. You can easily leave them out.
This'll give the foundation for your admins to create components that may not be as free as the proccall component but still able to run semi-decent events with it, unless you don't trust your admins to run events either.
This is very easy to accomplish, simply just remove most/all of the files from the admin component folder.

**Comment by ike709** [comment]
> This is very easy to accomplish, simply just remove most/all of the files from the admin component folder.

image

**Comment by Archanial** [comment]
How sure are you that this allows for elevation?
Admins have debug rights already

**Pull request closed without merging**